### PR TITLE
Move Variable.var and Variable.std to ATen

### PR DESCRIPTION
**Pull request by colesbury**

### CODE REVIEW

**Comment by colesbury** [comment]
Note that the ordering of parameters was inconsistent between Variable and Tensor. Tensor had:

```
std(torch.FloatTensor source, int dim, bool unbiased, bool keepdim)
```

Variable had:

```
std(torch.FloatTensor source, int dim, bool keepdim, bool unbiased)
```

This changes Variable to match the Tensor order

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -514,7 +514,11 @@
 - name: squeeze(Tensor self, int64_t dim)
   self: maybe_unsqueeze(grad, dim, self.size(dim) == 1 && self.sizes().size() != 1)
 
-- name: std
+- name: std(Tensor self, bool unbiased)
+  self: var_backward(grad * (result * 2).reciprocal(), self, unbiased)
+
+- name: std(Tensor self, int64_t dim, bool unbiased, bool keepdim)
+  self: var_backward(grad * (destination * 2).reciprocal(), self, dim, unbiased, keepdim)
```
Comment (gchanan) [inline tools/autograd/derivatives.yaml]: is destination the name of the return?  This magic sort of sucks.

**Review by gchanan — APPROVED** [review]
I didn't check the math, but the structure looks good.

File: tools/autograd/derivatives.yaml (hunk 1 above)
Comment (colesbury) [reply tools/autograd/derivatives.yaml]: Yes. Should we just name all the outputs some thing like `output` or `output[0]` for multiple returns?

Comment (gchanan) [reply tools/autograd/derivatives.yaml]: I'm ok with that or just allowing you to manually specify the name for cases like this in the format of native_functions.yaml, e.g.
```
- name: std(...) -> destination
```
or
```
- name: func_with_2_returns(...) -> max, indices
```
I'm not sure if that fits in with how we are doing the code gen, but at least it's not a dangling reference.

**Pull request merged**